### ISSUE TITLE
Terminal crashes when launched by `start /MAX`

### ISSUE THREAD

**Issue by pejons** [microsoft/terminal#17161]
### Windows Terminal version

1.19.10821.0

### Windows build number

10.0.22631.0

### Steps to reproduce

Create the following batch file:
```batch
@echo off

REM restart maximized if not already
if not "%1" == "max" start /MAX cmd /c %0 max & exit /b

echo Beep boop
timeout /t 20

```

With all Terminal windows closed, double-click to execute the script.

### Expected Behavior

Terminal window should always open maximized and display the 20 second countdown.

### Actual Behavior

You will see a terminal window pop up momentarily then exit, rather than waiting the 20 seconds.

Open a Terminal window in the background, then double-click to execute the script again. You will see a maximized terminal window pop up showing the countdown from 20 seconds (as expected).

Feedback Hub link: https://aka.ms/AAq7vl0

**Comment by zadjii-msft** [comment]
Thanks for filing!

* Do you have any memory if this worked in 1.18 (or earlier)/?
* Does something similar happen with the batch script in #14499?
* Does this work if you remove the `/max`, or if you change it to something like:
```cmd
@echo off

REM restart maximized if not already
if not "%1" == "max" start /MAX cmd /c %0 max

echo Beep boop
timeout /t 20
```

**Comment by pejons** [comment]
> Do you have any memory if this worked in 1.18 (or earlier)?

No, I started on the project where I ran into this very recently.

> Does something similar happen with the batch script in #14499?

No, that script seems to work as expected.

> Does this work if you remove the `/max`

Yes, the following script works without issue (except it doesn't maximize, of course)
```batch
@echo off

REM restart maximized if not already
if not "%1" == "max" start /MAX cmd /c %0 max & exit /b

echo Beep boop
timeout /t 20
```

> or if you change it to something like

Yes, that also works as expected (one non-maximized terminal window opens, followed by a maximized terminal window).

-----------

I also tried messing with the rending options, none of which made a difference:
- AtlasEngine enabled, hardware rendering
- AtlasEngine enabled, software rendering
- AtlasEngine disabled, software rendering

But with AtlasEngine disabled and hardware rendering, it seems like it takes longer to crash. And about 10% of the time, it wouldn't crash at all. Feels kinda like a race condition.

Finally, here is the original script which I minimized into the reproduction example. It doesn't crash, but does fail to minimize.
If I `REM` out the `call :echo-align center` line, it will crash as well.

```batch
@echo off

REM restart maximized if not already
if not "%1" == "max" start /MAX cmd /c %0 max & exit /b

title 202020

REM blank lines to center text vertically
for /L %%a in (1, 1, 30) do (
    echo.
)

REM message centered horizontally
call :echo-align center "look away for 20 seconds, press ESC to dismiss"

REM echo look away for 20 seconds, press ESC to dismiss

REM blank lines to push timeout countdown to bottom
for /L %%a in (1, 1, 20) do (
    echo.
)

timeout /t 20

:echo-align <align> <text>
	setlocal EnableDelayedExpansion
	(set^ tmp=%~2)
	if defined tmp (
		set "len=1"
		for %%p in (4096 2048 1024 512 256 128 64 32 16 8 4 2 1) do (
			if "!tmp:~%%p,1!" neq "" (
				set /a "len+=%%p"
				set "tmp=!tmp:~%%p!"
			)
		)
	) else (
		set len=0
	)

	for /f "skip=4 tokens=2 delims=:" %%i in ('mode con') do (
		set /a cols=%%i
		goto loop_end
	)
	:loop_end

	if /i "%1" equ "center" (
		set /a offsetnum=^(%cols% / 2^) - ^(%len% / 2^)
		set "offset="
		for /l %%i in (1 1 !offsetnum!) do set "offset=!offset! "
	) else if /i "%1" equ "right" (
		set /a offsetnum=^(%cols% - %len%^)
		set "offset="
		for /l %%i in (1 1 !offsetnum!) do set "offset=!offset! "
	)

	echo %offset%%~2
	endlocal

	exit /b
```

**Comment by zadjii-msft** [comment]
oh this might be related to #15936 

the first console process launches, attaches, launches a second, and before the second attaches the first detaches and the console closes but the terminal doesn't

And @lhecker has an idea how to fix this
